libs/hdf-converters/src/snyk-mapper.ts

```diff
@@ -37,142 +39,467 @@ function nistTag(identifiers: unknown[]): string[] {
+                }
+              },
+              refs: [],
+              source_location: {},
+              title: {
+                path: `${isApi?.apiVersion === 1 ? 'issueData.' : ''}title`
+              },
+              id: {path: 'id'},
+              desc: {
+                path: `${
+                  isApi?.apiVersion === 1 ? 'issueData.' : ''
+                }description`
+              },
+              impact: {
+                path: `${isApi?.apiVersion === 1 ? 'issueData.' : ''}severity`,
+                transformer: impactMapping(IMPACT_MAPPING)
+              },
+              code: {
+                transformer: (item: Record<string, unknown>): string => {
+                  return JSON.stringify(item, null, 2);
+                }
+              },
+              results: [
+                {
+                  status: ExecJSON.ControlResultStatus.Failed,
+                  code_desc: {
+                    ...(!isApi && {path: 'from'}),
+                    transformer: (input: unknown): string => {
+                      let paths = input;
+                      if (isApi?.apiVersion === 1) {
+                        paths = Object.values(
+                          _.pick(
+                            _.get(this.unconvertedData, 'dependencyChains'),
+                            (_.get(input, 'pkgVersions') as string[]).map(
+                              (version) =>
+                                `${_.get(input, 'pkgName')}@${version}`
+                            )
+                          )
+                        );
+                      }
+                      if (Array.isArray(paths) && paths.length > 0) {
+                        if (!Array.isArray(paths[0])) {
+                          paths = [[paths]];
+                        } else if (!Array.isArray(paths[0][0])) {
+                          paths = [paths];
+                        }
+                        return (paths as string[][][]) // versions/multiple paths/specific path
+                          .map((path) =>
+                            path
+                              .map((p) => `From : [ ${p.join(' , ')} ]`)
+                              .join('\n')
+                          )
+                          .join('\n');
+                      } else {
+                        return '';
+                      }
+                    }
+                  },
+                  start_time: ''
+                }
+              ]
+            }
+          ],
+          sha256: ''
         }
+      ],
+      passthrough: {
+        transformer: (data: Record<string, unknown>): Record<string, unknown> =>
+          isApi?.apiVersion === 1
+            ? {
+                raw: _.omit(data, [
+                  'organization',
+                  'project',
+                  'dependencyChains'
+                ])
+              }
+            : {
+                snyk_metadata: _.omit(data, ['vulnerabilities'])
+              }
```

Comment:
Raw is only gonna contain the original data (as converted into json) or the api calls.  The things that I am omitting there are additional attributes that I added to the data object based off of processing that I did.  

I'm renaming "metadata" to "unmapped_data" for lack of a better name atm, and that will contain the stuff that isn't mapped into hdf.  in the case of a file input, that means that it's like a handful of attributes or less since everything else is in the vulnerabilities array, which itself is completely mapped due to dumping each issue into the code attribute.

Reply:
Alright. Also, got a confirmation that the new naming scheme for unused data will be `other_source_tool_data` instead of `unmapped_data`.